src/test/regress/sql/multi_alter_table_add_constraints_without_name.sql

```diff
@@ -0,0 +1,187 @@
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+		WHERE rel.relname LIKE 'very%' ORDER BY con.conname ASC;
+
+-- Constraint can be deleted via the coordinator
+\c - - :master_host :master_port
+ALTER TABLE sc1.verylonglonglonglonglonglonglonglonglonglonglonglonglonglonglon DROP CONSTRAINT verylonglonglonglonglonglonglonglonglonglonglonglonglonglo_pkey;
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'very%';
+
+-- Test the scenario where a partitioned distributed table has a child with max allowed name
+-- Verify that we switch to sequential execution mode to avoid deadlock in this scenario
+\c - - :master_host :master_port
+CREATE TABLE dist_partitioned_table (dist_col int, another_col int, partition_col timestamp) PARTITION BY RANGE (partition_col);
+CREATE TABLE  p1 PARTITION OF dist_partitioned_table FOR VALUES FROM ('2021-01-01') TO ('2022-01-01');
+CREATE TABLE  longlonglonglonglonglonglonglonglonglonglonglonglonglonglongabc PARTITION OF dist_partitioned_table FOR VALUES FROM ('2020-01-01') TO ('2021-01-01');
+SELECT create_distributed_table('dist_partitioned_table', 'partition_col');
+
+SET client_min_messages TO DEBUG1;
+ALTER TABLE dist_partitioned_table ADD PRIMARY KEY(partition_col);
+RESET client_min_messages;
+
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname = 'dist_partitioned_table';
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'longlonglonglonglonglonglonglonglong%' ORDER BY con.conname ASC;
+
+\c - - :master_host :master_port
+ALTER TABLE dist_partitioned_table DROP CONSTRAINT dist_partitioned_table_pkey;
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'longlonglonglonglonglonglonglonglong%' ORDER BY con.conname ASC;
+
+-- Test primary key name is generated by postgres for citus local table.
+\c - - :master_host :master_port
+CREATE TABLE citus_local_table(id int, other_column int);
+ALTER TABLE citus_local_table ADD PRIMARY KEY(id);
```

Comment:
I believe this is not a Citus local table. The coordinator is not added as a worker node, which is a prerequisite. After that, citus.use_citus_managed_tables is not set, nor is citus_add_local_table_to_metadata UDF used.

Reply:
Fixed.